front/src/applications/stdcmV2/components/StdcmVias.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useMemo } from 'react';
+
+import { Location } from '@osrd-project/ui-icons';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+
+import IntermediatePointIcon from 'assets/pictures/stdcmV2/intermediate-point.svg';
+import { useOsrdConfSelectors, useOsrdConfActions } from 'common/osrdContext';
+import type { StdcmConfSliceActions } from 'reducers/osrdconf/stdcmConf';
+import type { PathStep } from 'reducers/osrdconf/types';
+import { useAppDispatch } from 'store';
+import { addElementAtIndex, replaceElementAtIndex } from 'utils/array';
+import { formatDurationAsISO8601 } from 'utils/timeManipulation';
+
+import StdcmCard from './StdcmCard';
+import StdcmInputVia from './StdcmInputVia';
+import StdcmOperationalPoint from './StdcmOperationalPoint';
+import type { StdcmSimulationResult } from '../views/StdcmViewV2';
+
+const StdcmVias = ({
+  disabled = false,
+  setCurrentSimulationInputs,
+}: {
+  disabled?: boolean;
+  setCurrentSimulationInputs: React.Dispatch<
+    React.SetStateAction<StdcmSimulationResult['input'] | undefined>
+  >;
+}) => {
+  const { t } = useTranslation('stdcm');
+  const dispatch = useAppDispatch();
+  const { getPathSteps } = useOsrdConfSelectors();
+  const { updatePathSteps, updateViaStopTimeV2 } = useOsrdConfActions() as StdcmConfSliceActions;
+  const pathSteps = useSelector(getPathSteps);
+
+  const intermediatePoints = useMemo(() => pathSteps.slice(1, -1), [pathSteps]);
+
+  const updatePathStepsList = (pathStep: PathStep | null, index: number) => {
+    const newPathSteps = replaceElementAtIndex(pathSteps, index, pathStep);
+    dispatch(updatePathSteps(newPathSteps));
+  };
+
+  const updatePathStepStopTime = (stopTime: string, index: number) => {
+    const pathStepToUpdate = pathSteps[index];
+    if (!pathStepToUpdate) return;
+    dispatch(
+      updateViaStopTimeV2({
+        via: pathStepToUpdate,
+        duration: formatDurationAsISO8601(Number(stopTime) * 60),
+      })
+    );
+  };
+
+  const deletePathStep = (index: number) => {
+    dispatch(updatePathSteps(pathSteps.filter((_, i) => i !== index)));
+  };
+
+  useEffect(() => {
+    setCurrentSimulationInputs((prevState) => ({
+      ...prevState,
+      pathSteps,
+    }));
+  }, [pathSteps]);
+
+  return (
+    <div className="stdcm-v2-vias-list">
+      {intermediatePoints.length > 0 &&
+        intermediatePoints.map((pathStep, index) => {
+          const pathStepId = index + 1;
+          return (
+            <StdcmCard
+              key={pathStepId}
+              name={t('trainPath.vias')}
+              title={
+                <div className="stdcm-v2-via-icons">
+                  <span>
+                    <img src={IntermediatePointIcon} alt="intermediate-point" />
+                  </span>
+                  <button type="button" onClick={() => deletePathStep(pathStepId)}>
```

Comment:
We have an action for that

```suggestion
                  <button type="button" onClick={() => dispatch(deleteViaV2(index))}>
```